File: dotnet-version/ParkingSystemDotnet/Services/ParkingService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using ParkingSystemDotnet.Models;

namespace ParkingSystemDotnet.Services
{
    public class ParkingService
    {
        private readonly ParkingLot _parkingLot;

        public ParkingService(ParkingLot parkingLot)
        {
            _parkingLot = parkingLot;
        }

        public void CountFreeSlots()
        {
            Console.WriteLine($"Slots free: {_parkingLot.Space - _parkingLot.Vehicles.Count}");
        }

        public void Status()
        {
            Console.WriteLine("Parking Status");
            if (_parkingLot.Vehicles.Any())
            {
                Console.WriteLine("Slot\t\tNo.\t\tType\t\tColor");
                foreach (var vehicle in _parkingLot.Vehicles)
                {
                    Console.WriteLine($"{vehicle.QueueNo}.\t\t{vehicle.PlateNo}\t\t{vehicle.Type}\t\t{vehicle.Color}");
                }
            }
            else Console.WriteLine("Parking Lot is empty");
        }

        public void Park(string details)
        {
            var info = details.Split(' ');
            int queue = Enumerable.Range(1, _parkingLot.Space)
                .FirstOrDefault(i => !_parkingLot.Vehicles.Any(v => v.QueueNo == i));

            if (queue != 0)
            {
                var vehicleType = info[2].Equals("mobil", StringComparison.OrdinalIgnoreCase) ? VehicleType.Mobil : VehicleType.Motor;
                var vehicle = new Vehicle(queue, info[0], info[1], vehicleType);
                _parkingLot.Vehicles.Add(vehicle);
                Console.WriteLine($"Allocated slot number: {vehicle.QueueNo}");
            }
            else Console.WriteLine("Parking lot is full");
        }

        public void Leave(int queueNo)
        {
            var vehicle = _parkingLot.Vehicles.FirstOrDefault(v => v.QueueNo == queueNo);
            if (vehicle != null)
            {
                _parkingLot.Vehicles.Remove(vehicle);
                Console.WriteLine($"Slot number {queueNo} is free");
            }
        }

        public void Search(int input, string param)
        {
            switch (input)
            {
                case 1:
                    GetVehiclesPlateNoByColor(param);
                    break;
                case 2:
                    GetVehiclesQueueNoByColor(param);
                    break;
                case 3:
                    GetVehiclesByOddPlate();
                    break;
                case 4:
                    GetVehiclesByEvenPlate();
                    break;
                case 5:
                    CountVehiclesByType(VehicleType.Mobil);
                    break;
                case 6:
                    CountVehiclesByType(VehicleType.Motor);
                    break;
                case 7:
                    GetSlotNumberByPlateNumber(param);
                    break;
            }
        }

        private void GetVehiclesQueueNoByColor(string color)
        {
            Console.WriteLine(string.Join(", ", _parkingLot.Vehicles.Where(v => v.Color.Equals(color, StringComparison.OrdinalIgnoreCase)).Select(v => v.QueueNo)));
        }

        private void GetVehiclesPlateNoByColor(string color)
        {
            Console.WriteLine(string.Join(", ", _parkingLot.Vehicles.Where(v => v.Color.Equals(color, StringComparison.OrdinalIgnoreCase)).Select(v => v.PlateNo)));
        }

        private void CountVehiclesByType(VehicleType type)
        {
            Console.WriteLine(_parkingLot.Vehicles.Count(v => v.Type == type));
        }

        private void GetSlotNumberByPlateNumber(string plateNumber)
        {
            var vehicle = _parkingLot.Vehicles.FirstOrDefault(v => v.PlateNo.Equals(plateNumber, StringComparison.OrdinalIgnoreCase));
            Console.WriteLine(vehicle != null ? vehicle.QueueNo.ToString() : "Plate number not found");
        }

        private void GetVehiclesByEvenPlate()
        {
            foreach (var vehicle in _parkingLot.Vehicles)
            {
                if (int.TryParse(vehicle.PlateNo.Split('-')[1], out int numbers) && numbers % 2 == 0)
                {
                    Console.WriteLine($"{vehicle.QueueNo}.\t{vehicle.PlateNo}");
                }
            }
        }

        private void GetVehiclesByOddPlate()
        {
            foreach (var vehicle in _parkingLot.Vehicles)
            {
                if (int.TryParse(vehicle.PlateNo.Split('-')[1], out int numbers) && numbers % 2 != 0)
                {
                    Console.WriteLine($"{vehicle.QueueNo}.\t{vehicle.PlateNo}");
                }
            }
        }
    }
}

File: src/service/ParkingService.java
package service;

import entity.ParkingLot;
import entity.Vehicle;
import entity.VehicleType;

public class ParkingService {
    private final ParkingLot parkingLot;

    public ParkingService(ParkingLot parkingLot) {
        this.parkingLot = parkingLot;
    }

    public void countFreeSlots() {
        System.out.println("Slots free: "+(parkingLot.getSpace()-parkingLot.getVehicles().size()));
    }

    public void status() {
        System.out.println("Parking Status");
        if (!parkingLot.getVehicles().isEmpty()) {
            System.out.println("Slot\t\tNo.\t\tType\t\tColor");
            for (Vehicle vehicle : parkingLot.getVehicles()) {
                System.out.println(vehicle.getQueueNo() + ".\t\t" + vehicle.getPlateNo() + "\t\t" + vehicle.getType() + "\t\t" + vehicle.getColor());
            }
        }else System.out.println("Parking Lot is empty");
    }

    public void park(String details) {
        String[] info = details.split(" ");
        int queue = 0;
        for (int i = 1; i <= parkingLot.getSpace(); i++) {
            boolean isEmpty = true;
            for (Vehicle vehicle:parkingLot.getVehicles()) {
                if (vehicle.getQueueNo()==i) {
                    isEmpty=false;
                    break;
                }
            }if (isEmpty) {
                queue=i;
                break;
            }
        }
        if (queue!=0) {
            Vehicle vehicle = new Vehicle(queue,info[0],info[1],info[2].equalsIgnoreCase("mobil")? VehicleType.MOBIL:VehicleType.MOTOR);
            parkingLot.getVehicles().add(vehicle);
            System.out.println("Allocated slot number: " + vehicle.getQueueNo());
        }else System.out.println("Parking lot is full");
    }

    public void leave(int queueNo) {
        for (Vehicle vehicle:parkingLot.getVehicles()) {
            if (vehicle.getQueueNo()==queueNo) {
                parkingLot.getVehicles().remove(vehicle);
            }
        }
        System.out.println("Slot number "+queueNo+" is free");
    }

    public void search(int input, String param) {
        switch (input) {
            case 1:
                getVehiclesPlateNoByColor(param);
                break;
            case 2:
                getVehiclesQueueNoByColor(param);
                break;
            case 3:
                getVehiclesByOddPlate();
                break;
            case 4:
                getVehiclesByEvenPlate();
                break;
            case 5:
                countVehiclesByType(VehicleType.MOBIL);
                break;
            case 6:
                countVehiclesByType(VehicleType.MOTOR);
                break;
            case 7:
                getSlotNumberByPlateNumber(param);
                break;
        }
    }
    public void getVehiclesQueueNoByColor(String color) {
        for (Vehicle vehicle:parkingLot.getVehicles()) {
            if (vehicle.getColor().equalsIgnoreCase(color))
                System.out.print(vehicle.getQueueNo()+", ");
        }
    }

    public void getVehiclesPlateNoByColor(String color) {
        for (Vehicle vehicle:parkingLot.getVehicles()) {
            if (vehicle.getColor().equalsIgnoreCase(color))
                System.out.print(vehicle.getPlateNo()+", ");
        }
    }

    public void countVehiclesByType(VehicleType type) {
        int count = 0;
        for (Vehicle vehicle:parkingLot.getVehicles()) {
            if (vehicle.getType().equals(type))
                count+=1;
        }
        System.out.println(count);
    }

    public void getSlotNumberByPlateNumber(String plateNumber) {
        boolean isExist = false;
        for (Vehicle vehicle:parkingLot.getVehicles()) {
            if (vehicle.getPlateNo().equalsIgnoreCase(plateNumber)) {
                System.out.println(vehicle.getQueueNo());
                isExist = true;
            }
        }
        if (!isExist)
            System.out.println("Plate number not found");
    }

    public void getVehiclesByEvenPlate() {
        for (Vehicle vehicle:parkingLot.getVehicles()) {
            String[] plate = vehicle.getPlateNo().split("-");
            int numbers = Integer.parseInt(plate[1]);
            if (numbers%2==0)
                System.out.println(vehicle.getQueueNo()+".\t"+vehicle.getPlateNo());
        }
    }

    public void getVehiclesByOddPlate() {
        for (Vehicle vehicle:parkingLot.getVehicles()) {
            String[] plate = vehicle.getPlateNo().split("-");
            int numbers = Integer.parseInt(plate[1]);
            if (numbers%2!=0)
                System.out.println(vehicle.getQueueNo()+".\t"+vehicle.getPlateNo());
        }
    }
}
